app/src/main/java/com/projectit3048c/MainActivity.kt

```diff
@@ -4,72 +4,166 @@ import android.os.Bundle
+        dropDownExpanded: Boolean,
+        list: List<Food>,
+        label: String = ""
+    ) {
+        Box(modifier) {
+            TextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .onFocusChanged { focusState ->
+                        if (!focusState.isFocused)
+                            onDismissRequest()
+                    },
+                value = value,
+                onValueChange = setValue,
+                label = { Text(label) },
+                colors = TextFieldDefaults.outlinedTextFieldColors()
+            )
+            DropdownMenu(
+                expanded = dropDownExpanded,
+                properties = PopupProperties(
+                    focusable = false,
+                    dismissOnBackPress = true,
+                    dismissOnClickOutside = true
+                ),
+                onDismissRequest = onDismissRequest
+            ) {
+                list.forEach { text ->
+                    DropdownMenuItem(onClick = {
+                        setValue(
+                            TextFieldValue(
+                                text.toString(),
+                                TextRange(text.toString().length)
+                            )
+                        )
+                    }) {
+                        Text(text = text.toString())
+                    }
                 }
             }
         }
     }
+
+
+
+
+
     @Composable
-    fun CalorieFacts(name:String) {
+    fun CalorieFacts(name:String, foods: List<Food> = ArrayList<Food>()) {
         var foodSearch by remember { mutableStateOf("") }
         var foodItem by remember { mutableStateOf("") }
         var itemCalorie by remember { mutableStateOf("") }
         var itemServing by remember { mutableStateOf("") }
         lateinit var food: Unit
         val context = LocalContext.current
         Column {
-            OutlinedTextField(
-                value = foodSearch,
-                onValueChange = { foodSearch = it },
-                label = { Text(stringResource(R.string.foodSearch)) },
-                modifier = Modifier.fillMaxWidth()
-            )
-            Button(
-                onClick = {
-                    food = viewModel.getSearchResult(foodSearch)
-                }
+            TextFieldWithDropdownUsage(dataIn = foods, stringResource(R.string.foodSearch))
 
-            ) {
-                Text(text = "Search")
-            }
+           // OutlinedTextField(
+                //value = foodSearch,
+                //onValueChange = { foodSearch = it },
+               // label = { Text(stringResource(R.string.foodSearch)) },
+               // modifier = Modifier.fillMaxWidth()
+            //)
+           // Button(
+               // onClick = {
+                   // food = viewModel.getSearchResult(foodSearch)
+               // }
+
+          //  ) {
+                //Text(text = "Search")
+           // }
```

Comment:
Avoid pushing commented code to Version Control.  Version control gives us better options:
- You can look at history to see what used to be there.
- You can use branches for experimental features.